Aliucord/src/main/java/com/aliucord/coreplugins/CoreCommands.kt

```diff
@@ -38,8 +49,13 @@ internal class CoreCommands : Plugin() {
             CommandResult(it.getRequiredString("message"))
         }
 
-        fun formatPlugins(plugins: List<Plugin>, showVersions: Boolean): String =
-            plugins.joinToString(transform = { p -> p.getName() + if (showVersions) " (${p.manifest.version})" else "" })
+        fun formatPlugins(plugins: List<Plugin>, showVersions: Boolean, joiner: String = ", ", showOutdated: Boolean = false): String =
+            plugins.joinToString(transform = { p ->
+                p.getName() + (if (showVersions) " (${p.manifest.version})" else "") + (if (showOutdated && PluginUpdater.checkPluginUpdate(
```

Comment:
this would be a lot more readable if you used StringBuilder

Reply:
```kt
StringBuilder().run {
    plugins.forEach {
        append(it.getName())
        if (showVersions) append(" (").append(it.manifest.version).append(')')
        if (showOutdated && PluginUpdate.checkPluginUpdate(it)) append(" (Outdated)")
        append(joiner)
    }
    setLength(length - joiner.length) // Remove last joiner
    toString()
}

```

Reply:
this looks worse imo but whatever